### ISSUE TITLE
Borders are not appeared on merged cells 

### ISSUE THREAD

**Issue by AAO2014** [kz26/PyExcelerate#74]
To make the borders to be visible I have to draw the borders before these cells will be merged. Otherwise, the borders will be drawn partly, only on some merged cells.

**Comment by kevmo314** [comment]
Thanks for the report, this one might be tricky to fix. Seems like the spec is a bit vague here, but we'll take a look.

**Comment by hamx0r** [comment]
I'm seeing a related border issue so I'm adding a note here instead of opening a new ticket:

I have 4 cells merged as a header and I'm adding Left borders to the left-most column of the 4, and a Right border to the right-most. The middle two columns are untouched style-wise.  However, I'm getting Right and Left columns applied to the "outside" two columns like this:
![extra_borders](https://user-images.githubusercontent.com/423974/44593094-58f1f980-a777-11e8-8084-c133097030d4.jpg)

The code producing this is:
                            # Set vertical borders bounding cols D through G for 29 rows 
                            start = "D1
                            end = "G1"
                            ws.range(start, end).merge()

                            # Set vertical borders
                            start = "D1"
                            end = "D29"
                            ws.range(start, end).style.borders.left.style = '_'

                            start = "G1"
                            end = "G29"
                            ws.range(start, end).style.borders.right.style = '_'

                            # Now set vertical borders bounding cols H through K 
                            start = "H1
                            end = "K1"
                            ws.range(start, end).merge()

                            # Set vertical borders
                            start = "H1"
                            end = "H29"
                            ws.range(start, end).style.borders.left.style = '_'

                            start = "K1"
                            end = "K29"
                            ws.range(start, end).style.borders.right.style = '_'

If I instead draw borders starting from row 2 (ie change `start` to D2, G2, H2 and K2 respectively above) then I no longer get the Right border on cols D & H, but _still_ get Left borders on cols G & K.  This tells me that the merged cells in row 1 are having an impact.  If I remove the code merging those cells, or place it _after_ the code drawing borders, then I get the results in this paragraph (ie only get the extra Left border in cols G&K).   

To me, it's a fine solution to obey the rule "Draw borders before merging".   I'd still like the borders to be drawn for the ranges defined and not have extra borders added in on G and K. 

The expected borders are here (notice no right borders on cols D & H, nor left borders on cols G & K)
![expected_borders](https://user-images.githubusercontent.com/423974/44593437-2dbbda00-a778-11e8-92a6-fc99436ffd50.jpg)

**Comment by kevmo314** [comment]
Yeah the spec is kind of vague about how to accurately specify this kind of situation. I'll take a look but it'll require a bit of experimentation to see what kind of configurations Excel spits out, so this may be a bit challenging to solve effectively.
